File: server/src/main/java/me/ajfleming/tworoomsio/service/lobby/JoinCodeGenerator.java
package me.ajfleming.tworoomsio.service.lobby;

import java.util.Locale;
import org.apache.commons.lang3.RandomStringUtils;
import org.springframework.stereotype.Component;

@Component
public class JoinCodeGenerator {

  public String generateJoinCode() {
    return RandomStringUtils.randomAlphabetic(4).toUpperCase(Locale.ROOT);
  }
}

File: kotlin_server/src/main/kotlin/me/ajfleming/tworoomsio/service/JoinCodeGenerator.kt
package me.ajfleming.tworoomsio.service

class JoinCodeGenerator {
    companion object {
        private const val JOIN_CODE_LENGTH = 4

        fun generateJoinCode() : String {
            val allowedChars = ('A'..'Z')
            return (1..JOIN_CODE_LENGTH)
                .map { allowedChars.random() }
                .joinToString { "" }
        }
    }
}